x/protorev/keeper/developer_fees_test.go

```diff
@@ -0,0 +1,168 @@
+				err = suite.App.AppKeepers.ProtoRevKeeper.UpdateDeveloperFees(suite.Ctx, sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(1000)), sdk.NewInt(2000))
+			},
+			expectedErr:   false,
+			expectedCoins: sdk.NewCoins(sdk.NewCoin(types.AtomDenomination, sdk.NewInt(200)), sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(350))),
+		},
+	}
+
+	for _, tc := range cases {
+		suite.Run(tc.description, func() {
+			suite.SetupTest()
+			tc.malleate()
+
+			err := suite.App.AppKeepers.ProtoRevKeeper.SendDeveloperFeesToDeveloperAccount(suite.Ctx)
+			if tc.expectedErr {
+				suite.Require().Error(err)
+			} else {
+				suite.Require().NoError(err)
+			}
+
+			developerAccount, err := suite.App.AppKeepers.ProtoRevKeeper.GetDeveloperAccount(suite.Ctx)
+			if err == nil {
+				developerFees := suite.App.AppKeepers.BankKeeper.GetAllBalances(suite.Ctx, developerAccount)
+				suite.Require().Equal(tc.expectedCoins, developerFees)
+			}
+		})
+	}
+}
+
+// TestUpdateDeveloperFees tests the UpdateDeveloperFees function
+func (suite *KeeperTestSuite) TestUpdateDeveloperFees() {
+	cases := []struct {
+		description    string
+		inputCoin      sdk.Coin
+		tokenOutAmount sdk.Int
+		malleate       func()
+		expected       sdk.Coin
+	}{
+		{
+			description:    "Update developer fees in year 1",
+			inputCoin:      sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(100)),
+			tokenOutAmount: sdk.NewInt(200),
+			malleate:       func() {},
+			expected:       sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(20)),
+		},
+		{
+			description:    "Update developer fees in year 2",
+			inputCoin:      sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(100)),
+			tokenOutAmount: sdk.NewInt(200),
+			malleate: func() {
+				suite.App.AppKeepers.ProtoRevKeeper.SetDaysSinceGenesis(suite.Ctx, 366)
+			},
+			expected: sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(10)),
+		},
+		{
+			description:    "Update developer fees after year 2",
+			inputCoin:      sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(100)),
+			tokenOutAmount: sdk.NewInt(200),
+			malleate: func() {
+				suite.App.AppKeepers.ProtoRevKeeper.SetDaysSinceGenesis(suite.Ctx, 731)
+
+			},
+			expected: sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(5)),
+		},
+		{
+			description:    "Update developer fees after year 10",
+			inputCoin:      sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(100)),
+			tokenOutAmount: sdk.NewInt(200),
+			malleate: func() {
+				suite.App.AppKeepers.ProtoRevKeeper.SetDaysSinceGenesis(suite.Ctx, 365*10+1)
+
+			},
+			expected: sdk.NewCoin(types.OsmosisDenomination, sdk.NewInt(5)),
+		},
+	}
+
+	for _, tc := range cases {
+		suite.Run(tc.description, func() {
+			suite.SetupTest()
+			tc.malleate()
+
+			err := suite.App.AppKeepers.ProtoRevKeeper.UpdateDeveloperFees(suite.Ctx, tc.inputCoin, tc.tokenOutAmount)
+			suite.Require().NoError(err)
+
+			developerFees, err := suite.App.AppKeepers.ProtoRevKeeper.GetDeveloperFees(suite.Ctx, tc.inputCoin.Denom)
```

Comment:
Is the `feesDenom` always going to either osmo or atom?

Reply:
Yup. `x/protorev` exclusively executes arbitrage trades using either Osmo or Atom, so the fees sent to the developer account will always be denominated in one of those two assets.